zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/SQLPPGenerator.scala

```diff
@@ -334,6 +354,13 @@ class SQLPPGenerator extends AsterixQueryGenerator {
       case _ => aggFuncExpr(typeImpl.getAggregateStr(aggr.func))
     }
   }
+
+  protected def append(queryBuilder: StringBuilder, queryStr: String): Unit = {
+    if (!queryStr.isEmpty) {
```

Comment:
And add a comment to this function explain why we don't need to append. IMO it's more like a trick to write less code. :-)